regex-capi/src/lib.rs

```diff
@@ -0,0 +1,515 @@
+extern crate libc;
+extern crate regex;
+
+use std::collections::HashMap;
+use std::ffi::{CStr, CString};
+use std::fmt;
+use std::io::{self, Write};
+use std::ops::Deref;
+use std::panic::{self, AssertUnwindSafe};
+use std::process;
+use std::ptr;
+use std::slice;
+use std::str;
+
+use libc::{c_char, uint32_t};
+use regex::bytes;
+
+macro_rules! protect {
```

Comment:
Nice, thank you. I did this. It'd be cool to just have one `ffi` block and stuff all the extern fns in it, but I think that makes the macro much more complex.